src/Surfnet/ServiceProviderDashboard/Application/Service/ChangeRequestService.php

```diff
@@ -35,11 +37,18 @@ public function __construct(
     }
 
     /**
-     * @throws \Exception
+     * @param string $id
+     * @param Protocol $protocol
+     * @return ChangeRequestDtoCollection|Exception
      */
-    public function findById(string $id): ChangeRequestDtoCollection
+    public function findByIdAndProtocol(string $id, Protocol $protocol): ChangeRequestDtoCollection
     {
-        $values = $this->repository->getChangeRequest($id);
+        $values = [];
+        try {
+            $values = $this->repository->getChangeRequest($id, $protocol);
+        } catch (Exception $exception) {
+            throwException($exception);
```

Comment:
1. This construction seems irrelevant. You catch an exception and again throw it. Which is basically a no-operation.
2. Also, using `throwException` (a function I'm unaware of) is not in line with the way we are used to throw exceptions (just use `throw $exception` or `throw new FoobarException('Rudy, a message to you, Rudy')`
3. The find method should not throw an exception, but return an empty result if the data is not found. (getter throws exceptions, find methods allow for returning a null result.)